server/src/main/java/io/crate/types/NumericType.java

```diff
@@ -0,0 +1,232 @@
+    }
+
+    public static NumericType of(int precision, int scale) {
+        return new NumericType(precision, scale);
+    }
+
+    public static DataType<?> of(List<Integer> parameters) {
+        if (parameters.isEmpty() || parameters.size() > 2) {
+            throw new IllegalArgumentException(
+                "The numeric type support one or two parameter arguments, received: " +
+                parameters.size()
+            );
+        }
+        if (parameters.size() == 1) {
+            return of(parameters.get(0));
+        } else {
+            return of(parameters.get(0), parameters.get(1));
+        }
+    }
+
+    private final Integer scale;
+    private final Integer precision;
+
+    private NumericType(Integer precision, Integer scale) {
+        this.precision = precision;
+        this.scale = scale;
+    }
+
+    public NumericType(StreamInput in) throws IOException {
+        this.precision = in.readOptionalVInt();
+        this.scale = in.readOptionalVInt();
+    }
+
+    @Override
+    public int id() {
+        return ID;
+    }
+
+    @Override
+    public Precedence precedence() {
+        return Precedence.NUMERIC;
+    }
+
+    @Override
+    public String getName() {
+        return "numeric";
+    }
+
+    @Override
+    public Streamer<BigDecimal> streamer() {
+        return this;
+    }
+
+    @Override
+    public BigDecimal implicitCast(Object value) throws IllegalArgumentException, ClassCastException {
+        if (value == null) {
+            return null;
+        }
+
+        var mathContext = mathContextOrDefault();
+        if (value instanceof Long
+            || value instanceof Byte
+            || value instanceof Integer
+            || value instanceof Short) {
+            return new BigDecimal(
+                BigInteger.valueOf(((Number) value).longValue()),
+                mathContext
+            ).setScale(scaleOrDefault(), mathContext.getRoundingMode());
+        } else if (value instanceof String
+                   || value instanceof Float
+                   || value instanceof Double
+                   || value instanceof BigDecimal) {
+            return new BigDecimal(
+                value.toString(),
+                mathContext
+            ).setScale(scaleOrDefault(), mathContext.getRoundingMode());
+        } else {
+            throw new ClassCastException("Can't cast '" + value + "' to " + getName());
+        }
+    }
+
+    @Override
+    public BigDecimal sanitizeValue(Object value) {
+        if (value == null) {
+            return null;
+        } else {
+            return (BigDecimal) value;
+        }
+    }
+
+    @Override
+    public BigDecimal valueForInsert(Object value) {
+        throw new UnsupportedOperationException(
+            getName() + " type cannot be used in insert statements");
+    }
+
+    public static long size(@Nonnull BigDecimal value) {
+        return value.unscaledValue().bitLength() / 8 + 1;
```

Comment:
Can you please elaborate this size computation? Taking all fields of a BigDecimal into account, the minimal usage is already 36 bytes + required bits for the internal integers...